test/runnable/auto_ref.d

```diff
@@ -0,0 +1,55 @@
+struct S { }
+
+bool isLvalue(S s) {
+    return false;
+}
+
+bool isLvalue(ref S s) {
+    return true;
+}
+
+void oof(         S s) { }
+void ofo(     ref S s) { }
+void foo(auto ref S s) { }
+
+int bar(auto ref S s) { return 1; }
+//int bar(     ref S s) { return 1; }
+int bar(         S s) { return 2; }
```

Comment:
But that would mean that the auto ref function would be useless, since by ref and by value always have priority.